### ISSUE TITLE
Global TS constants/functions are not defined in JS scripts

### ISSUE THREAD

**Issue by zaunermax** [ioBroker/ioBroker.javascript#687]
**Describe the bug**  
I use global files, to store object ids to being able to switch them out in one place and use the constant in all my scripts.

My global files are all Typescript.

However after updating to the script engine 4.9.4 all global constants do not work anymore.

**To Reproduce**  
1. Have a TypeScript file in global with a constant/function
2. Add a JavaScript script to any folder
3. Try to use the constant in the startup of the script. (Creating states...)

**Expected behavior**  
The constants should be available on script startup.

**Versions:**  
 - Adapter version: 4.9.4
 - JS-Controller version: 3.1.5
 - Node version: 10.20.0
 - Operating system: docker - debian

**Additional context**  
The downgrade to 3.8.* did fix my problem.

**Comment by AlCalzone** [comment]
Please share a combination of scripts that causes the problem

**Comment by AlCalzone** [comment]
I can't reproduce with this setup:
![grafik](https://user-images.githubusercontent.com/17641229/97145657-dace7a80-1766-11eb-968d-b7a46557e21a.png)

**Global TS script:**  
```ts
function foo(): void {
    log("foo");
}
const baz = 1;
```

**Normal JS script:**
```js
foo();
log(`baz is ${baz}`);
```

**Log output:**
```
08:39:29.176 | info | javascript.0 (19516) Start javascript script.js.Test1
08:39:29.178 | info | javascript.0 (19516) script.js.Test1: foo
08:39:29.178 | info | javascript.0 (19516) script.js.Test1: baz is 1
08:39:29.178 | info | javascript.0 (19516) script.js.Test1: registered 0 subscriptions and 0 schedules
```

**Comment by winnyschuster** [comment]
>I can't reproduce with this setup:

i can with your scripts and get a reference error
`
javascript.0 | 2020-10-26 16:31:19.427 | error | (13574) at Script.runInContext (vm.js:142:18)
-- | -- | -- | --
javascript.0 | 2020-10-26 16:31:19.427 | error | (13574) at script.js.Aktiv.ScriptJS:3:3
javascript.0 | 2020-10-26 16:31:19.426 | error | (13574) at script.js.Aktiv.ScriptJS:1:1
javascript.0 | 2020-10-26 16:31:19.426 | error | (13574) script.js.Aktiv.ScriptJS: ReferenceError: foo is not defined
javascript.0 | 2020-10-26 16:31:19.423 | info | (13574) script.js.Aktiv.ScriptJS: registered 0 subscriptions and 0 schedules`


version 4.9.4
node 14.14.0

**Comment by AlCalzone** [comment]
Hmm weird. Can you enable expert mode in the objects and show me the global script object?
![grafik](https://user-images.githubusercontent.com/17641229/97195709-f14af500-17ab-11eb-93ac-e66a74ac8f40.png)
then press on the edit button on the right --> raw tab
![grafik](https://user-images.githubusercontent.com/17641229/97195791-06c01f00-17ac-11eb-8839-0d94efccdead.png)

**Comment by winnyschuster** [comment]
> Hmm weird. Can you enable expert mode in the objects and show me the global script object?

looks different from yours
`
{
  "common": {
    "name": "ts1",
    "expert": true,
    "engineType": "TypeScript/ts",
    "engine": "system.adapter.javascript.0",
    "source": "function foo(): void {\r\n    log(\"foo\");\r\n}\r\nconst baz = 1;",
    "debug": false,
    "verbose": false,
    "enabled": true,
    "sourceHash": "d9f07124d2404f101bef2a14efa97f4b",
    "compiled": "\"use strict\";\r\nvar __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {\r\n    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }\r\n    return new (P || (P = Promise))(function (resolve, reject) {\r\n        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }\r\n        function rejected(value) { try { step(generator[\"throw\"](value)); } catch (e) { reject(e); } }\r\n        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }\r\n        step((generator = generator.apply(thisArg, _arguments || [])).next());\r\n    });\r\n};\r\nvar __generator = (this && this.__generator) || function (thisArg, body) {\r\n    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;\r\n    return g = { next: verb(0), \"throw\": verb(1), \"return\": verb(2) }, typeof Symbol === \"function\" && (g[Symbol.iterator] = function() { return this; }), g;\r\n    function verb(n) { return function (v) { return step([n, v]); }; }\r\n    function step(op) {\r\n        if (f) throw new TypeError(\"Generator is already executing.\");\r\n        while (_) try {\r\n            if (f = 1, y && (t = op[0] & 2 ? y[\"return\"] : op[0] ? y[\"throw\"] || ((t = y[\"return\"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;\r\n            if (y = 0, t) op = [op[0] & 2, t.value];\r\n            switch (op[0]) {\r\n                case 0: case 1: t = op; break;\r\n                case 4: _.label++; return { value: op[1], done: false };\r\n                case 5: _.label++; y = op[1]; op = [0]; continue;\r\n                case 7: op = _.ops.pop(); _.trys.pop(); continue;\r\n                default:\r\n                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }\r\n                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }\r\n                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }\r\n                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }\r\n                    if (t[2]) _.ops.pop();\r\n                    _.trys.pop(); continue;\r\n            }\r\n            op = body.call(thisArg, _);\r\n        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }\r\n        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };\r\n    }\r\n};\r\nObject.defineProperty(exports, \"__esModule\", { value: true });\r\n(function () { return __awaiter(void 0, void 0, void 0, function () {\r\n    function foo() {\r\n        log(\"foo\");\r\n    }\r\n    var baz;\r\n    return __generator(this, function (_a) {\r\n        baz = 1;\r\n        return [2 /*return*/];\r\n    });\r\n}); })();\r\n",
    "declarations": "export {};\r\n"
  },
  "type": "script",
  "from": "system.adapter.admin.0",
  "user": "system.user.admin",
  "ts": 1603726082391,
  "_id": "script.js.global.ts1",
  "acl": {
    "object": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  },
  "native": {}
}

`

**Comment by winnyschuster** [comment]
in addition, i observed the following behavior: Once i change a script in global, the instance is restarting immediately. It doesnt matter if it is TS or JS, setting instance to debug mode does not show useful information, see below(sentry plugin was enabled too). Also, except for this test i never used global scripts or TS, all scripts i normally use run just fine. let me know if you need anything else
`
javascript.2 | 2020-10-26 19:59:06.011 | info | (2191) received all states
javascript.2 | 2020-10-26 19:59:05.816 | info | (2191) received all objects
javascript.2 | 2020-10-26 19:59:00.445 | info | (2191) requesting all objects
javascript.2 | 2020-10-26 19:59:00.442 | info | (2191) requesting all states
javascript.2 | 2020-10-26 19:59:00.441 | debug | (2191) Loaded TypeScript definitions for tr-O64: []
javascript.2 | 2020-10-26 19:59:00.437 | debug | (2191) Loaded TypeScript definitions for request: ["node_modules/@types/request/package.json","node_modules/@types/request/index.d.ts"]
javascript.2 | 2020-10-26 19:59:00.354 | debug | (2191) Loaded TypeScript definitions for node: ["node_modules/@types/node/package.json","node_modules/@types/node/index.d.ts","node_modules/@types/node/base.d.ts","node_modules/@types/node/ts3.6/base
javascript.2 | 2020-10-26 19:59:00.256 | info | (2191) starting. Version 4.9.4 in /opt/iobroker/node_modules/iobroker.javascript, node: v14.14.0, js-controller: 3.1.6
javascript.2 | 2020-10-26 19:58:59.114 | debug | (2191) Plugin sentry Initialize Plugin (enabled=true)
javascript.2 | 2020-10-26 19:58:58.550 | debug | (2191) statesDB connected
javascript.2 | 2020-10-26 19:58:58.549 | debug | (2191) States connected to redis: 127.0.0.1:9000
javascript.2 | 2020-10-26 19:58:58.525 | debug | (2191) States create System PubSub Client
javascript.2 | 2020-10-26 19:58:58.521 | debug | (2191) States create User PubSub Client
javascript.2 | 2020-10-26 19:58:58.502 | debug | (2191) Redis States: Use Redis connection: 127.0.0.1:9000
javascript.2 | 2020-10-26 19:58:58.496 | debug | (2191) objectDB connected
javascript.2 | 2020-10-26 19:58:58.475 | debug | (2191) Objects connected to redis: 127.0.0.1:9001
javascript.2 | 2020-10-26 19:58:58.439 | debug | (2191) Objects client initialize lua scripts
javascript.2 | 2020-10-26 19:58:58.438 | debug | (2191) Objects create User PubSub Client
javascript.2 | 2020-10-26 19:58:58.436 | debug | (2191) Objects create System PubSub Client
javascript.2 | 2020-10-26 19:58:58.431 | debug | (2191) Objects client ready ... initialize now
javascript.2 | 2020-10-26 19:58:58.386 | debug | (2191) Redis Objects: Use Redis connection: 127.0.0.1:9001
javascript.2 | 2020-10-26 19:58:52.372 | info | (1895) Terminated (ADAPTER_REQUESTED_TERMINATION): Without reason
javascript.2 | 2020-10-26 19:58:52.371 | debug | (1895) Plugin sentry destroyed
javascript.2 | 2020-10-26 19:58:52.369 | info | (1895) terminating
javascript.2 | 2020-10-26 19:58:52.360 | info | (1895) Got terminate signal TERMINATE_YOURSELF


`

**Comment by AlCalzone** [comment]
Yeah that restart is expected. At least it has always been this way, because changing a global script changes all scripts.

The other thing I need to look at. It seems like one of my fixes is missing from 4.9.4.

**Comment by AlCalzone** [comment]
Okay I found it:
https://github.com/ioBroker/ioBroker.javascript/commit/d5a1a06aa1d2e4a7f47868205db8e2374c003311 is merged but not yet released.

**Comment by zaunermax** [comment]
Thank you for fixing this so quickly :raised_hands: 
I just wanted to write that I have time tonight to look into what you wrote but I guess it's fixed already :rocket:

**Comment by AlCalzone** [comment]
should be fixed in 4.9.5

**Comment by winnyschuster** [comment]
> should be fixed in 4.9.5

it is :-)

**Comment by neopholus** [comment]
I have exactly the same problem.
Worked until 4.9.8, does not work since 4.10.1 anymore.
